tests/test_types.py

```diff
@@ -187,3 +363,90 @@ def test_as_jsonschema():
             }
         }
     }
+
+
+def test_extended_as_jsonschema_flat():
+    """
+    When allOf is implemented in apistar, the struct should look more like the following, give or take the "type"
+
+    assert struct == {
+        "$ref": "#/definitions/ReviewedProduct",
+        "definitions": {
+            "Product": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string",
+                        "maxLength": 10
+                    },
+                    "rating": {
+                        "type": "integer",
+                        "minimum": 0,
+                        "maximum": 100,
+                        "default": None,
+                        "nullable": True
+                    },
+                    "created": {
+                        "type": "string",
+                        "format": "datetime"
+                    }
+                },
+                "required": [
+                    "name",
+                    "created"
+                ]
+            },
+            "ReviewedProduct": {
+                "allOf": {
+                    "$ref": "#/definitions/Product"
+                },
+                "type": "object",
+                "properties": {
+                    "reviewer": {
+                        "type": "string",
+                        "maxLength": 20
+                    }
+                },
+                "required": [
+                    "reviewer"
+                ]
+            }
+        }
+    }
+
+    """
```

Comment:
I'd suggest we drop the docstring here. The behavior's currently good as it is, we don't need to document what we may intend to do in the future.